Some tests fail when run individually with error "ValueError: Level 'RED' does not exist"

When ran individually, all the tests in the `tests/attack` folder fail with this error:

```
ValueError: Level 'RED' does not exist
```

This can be reproduced with:

```
(venv) wapiti ❯ pytest tests/attack/test_mod_buster.py 
====================================================================================================================== test session starts ======================================================================================================================
platform linux -- Python 3.9.6, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/polyedre/code/wapiti
plugins: cov-2.12.1, anyio-3.3.0, asyncio-0.14.0
collected 1 item                                                                                                                                                                                                                                                

tests/attack/test_mod_buster.py F                                                                                                                                                                                                                         [100%]

=========================================================================================================================== FAILURES ============================================================================================================================
_______________________________________________________________________________________________________________________ test_whole_stuff ________________________________________________________________________________________________________________________

    @pytest.mark.asyncio
    @respx.mock
    async def test_whole_stuff():
        # Test attacking all kind of parameter without crashing
        respx.get("http://perdu.com/").mock(return_value=httpx.Response(200, text="Default page"))
        respx.get("http://perdu.com/admin").mock(
            return_value=httpx.Response(301, text="Hello there", headers={"Location": "/admin/"})
        )
        respx.get("http://perdu.com/admin/").mock(return_value=httpx.Response(200, text="Hello there"))
        respx.get("http://perdu.com/config.inc").mock(return_value=httpx.Response(200, text="pass = 123456"))
        respx.get("http://perdu.com/admin/authconfig.php").mock(return_value=httpx.Response(200, text="Hello there"))
        respx.get(url__regex=r"http://perdu\.com/.*").mock(return_value=httpx.Response(404))
    
        persister = Mock()
    
        request = Request("http://perdu.com/")
        request.path_id = 1
        request.set_headers({"content-type": "text/html"})
        # Buster module will get requests from the persister
        persister.get_links.return_value = AsyncIterator([request])
    
        crawler = AsyncCrawler("http://perdu.com/", timeout=1)
        options = {"timeout": 10, "level": 2, "tasks": 20}
    
        with patch(
                "wapitiCore.attack.mod_buster.mod_buster.payloads",
                [("nawak", Flags()), ("admin", Flags()), ("config.inc", Flags()), ("authconfig.php", Flags())]
        ):
            module = mod_buster(crawler, persister, options, Event())
            module.verbose = 2
            module.do_get = True
>           await module.attack(request)

tests/attack/test_mod_buster.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
wapitiCore/attack/mod_buster.py:143: in attack
    await self.test_directory(current_dir)
wapitiCore/attack/mod_buster.py:114: in test_directory
    await task
wapitiCore/attack/mod_buster.py:59: in check_path
    self.log_red("Found webpage {0}", loc)
venv/lib/python3.9/site-packages/loguru/_logger.py:1997: in log
    __self._log(level_id, static_level_no, False, __self._options, __message, args, kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <loguru.logger handlers=[(id=0, level=10, sink=<_io.FileIO name=8 mode='rb+' closefd=True>)]>, level_id = 'RED', static_level_no = None, from_decorator = False, options = (None, 0, False, False, False, False, ...), message = 'Found webpage {0}'
args = ('http://perdu.com/admin/',), kwargs = {}

    def _log(self, level_id, static_level_no, from_decorator, options, message, args, kwargs):
        core = self._core
    
        if not core.handlers:
            return
    
        (exception, depth, record, lazy, colors, raw, capture, patcher, extra) = options
    
        frame = get_frame(depth + 2)
    
        try:
            name = frame.f_globals["__name__"]
        except KeyError:
            name = None
    
        try:
            if not core.enabled[name]:
                return
        except KeyError:
            enabled = core.enabled
            if name is None:
                status = core.activation_none
                enabled[name] = status
                if not status:
                    return
            else:
                dotted_name = name + "."
                for dotted_module_name, status in core.activation_list:
                    if dotted_name[: len(dotted_module_name)] == dotted_module_name:
                        if status:
                            break
                        enabled[name] = False
                        return
                enabled[name] = True
    
        current_datetime = aware_now()
    
        if level_id is None:
            level_icon = " "
            level_no = static_level_no
            level_name = "Level %d" % level_no
        else:
            try:
                level_name, level_no, _, level_icon = core.levels[level_id]
            except KeyError:
>               raise ValueError("Level '%s' does not exist" % level_id) from None
E               ValueError: Level 'RED' does not exist

venv/lib/python3.9/site-packages/loguru/_logger.py:1886: ValueError
--------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------
2021-08-22 18:41:15.339 | INFO     | wapitiCore.attack.mod_buster:test_directory:74 - [¨] Testing directory http://perdu.com/
======================================================================================================================= warnings summary ========================================================================================================================
wapitiCore/language/language.py:62
wapitiCore/language/language.py:62
  /home/polyedre/code/wapiti/wapitiCore/language/language.py:62: DeprecationWarning: parameter codeset is deprecated
    lan = gettext.translation(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
==================================================================================================================== short test summary info ====================================================================================================================
FAILED tests/attack/test_mod_buster.py::test_whole_stuff - ValueError: Level 'RED' does not exist
================================================================================================================= 1 failed, 2 warnings in 0.36s =================================================================================================================
```

It seems that the colored logging level are defined in `wapitiCore/main/wapiti.py` and that this file is not loaded when running the tests.

https://github.com/wapiti-scanner/wapiti/blob/c0aa358b84960c63c7c28db3e0031dab5d3261ea/wapitiCore/main/wapiti.py#L55-L59

--- a/wapitiCore/attack/attack.py
+++ b/wapitiCore/attack/attack.py
@@ -29,7 +29,7 @@ from binascii import hexlify
 from functools import partialmethod, partial
 
 from httpx import ReadTimeout, RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.language.vulnerability import CRITICAL_LEVEL, HIGH_LEVEL, MEDIUM_LEVEL, LOW_LEVEL, INFO_LEVEL
 from wapitiCore.net.web import Request

--- a/wapitiCore/attack/mod_backup.py
+++ b/wapitiCore/attack/mod_backup.py
@@ -25,7 +25,7 @@ from os.path import splitext
 from urllib.parse import urljoin
 
 from httpx import RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack
 from wapitiCore.language.vulnerability import LOW_LEVEL, _

--- a/wapitiCore/attack/mod_buster.py
+++ b/wapitiCore/attack/mod_buster.py
@@ -20,7 +20,7 @@
 import asyncio
 
 from httpx import RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack
 from wapitiCore.net.web import Request

--- a/wapitiCore/attack/mod_crlf.py
+++ b/wapitiCore/attack/mod_crlf.py
@@ -19,7 +19,7 @@
 from urllib.parse import quote
 
 from httpx import ReadTimeout, HTTPStatusError, RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack, Flags
 from wapitiCore.language.vulnerability import Messages, _

--- a/wapitiCore/attack/mod_exec.py
+++ b/wapitiCore/attack/mod_exec.py
@@ -17,7 +17,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 from httpx import ReadTimeout, RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack, PayloadType
 from wapitiCore.language.vulnerability import Messages, _

--- a/wapitiCore/attack/mod_file.py
+++ b/wapitiCore/attack/mod_file.py
@@ -22,7 +22,7 @@ from collections import defaultdict, namedtuple
 import re
 
 from httpx import ReadTimeout, RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack, PayloadReader
 from wapitiCore.language.vulnerability import Messages, _

--- a/wapitiCore/attack/mod_methods.py
+++ b/wapitiCore/attack/mod_methods.py
@@ -17,7 +17,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 from httpx import RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack
 from wapitiCore.definitions.methods import NAME

--- a/wapitiCore/attack/mod_nikto.py
+++ b/wapitiCore/attack/mod_nikto.py
@@ -24,7 +24,7 @@ import random
 from urllib.parse import urlparse
 
 from httpx import RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack
 from wapitiCore.language.vulnerability import _

--- a/wapitiCore/attack/mod_permanentxss.py
+++ b/wapitiCore/attack/mod_permanentxss.py
@@ -21,7 +21,7 @@ from configparser import ConfigParser
 from os.path import join as path_join
 
 from httpx import ReadTimeout, RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack, PayloadType, Mutator
 from wapitiCore.language.vulnerability import Messages, _

--- a/wapitiCore/attack/mod_redirect.py
+++ b/wapitiCore/attack/mod_redirect.py
@@ -17,7 +17,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 from httpx import RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack, Flags
 from wapitiCore.language.vulnerability import Messages, _

--- a/wapitiCore/attack/mod_sql.py
+++ b/wapitiCore/attack/mod_sql.py
@@ -20,7 +20,7 @@ import re
 from random import randint
 
 from httpx import ReadTimeout, RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack, Flags, Mutator
 from wapitiCore.language.vulnerability import Messages, _

--- a/wapitiCore/attack/mod_ssrf.py
+++ b/wapitiCore/attack/mod_ssrf.py
@@ -21,7 +21,7 @@ from urllib.parse import quote
 from binascii import hexlify, unhexlify
 
 from httpx import RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack, Mutator, PayloadType, Flags
 from wapitiCore.language.vulnerability import Messages, _

--- a/wapitiCore/attack/mod_timesql.py
+++ b/wapitiCore/attack/mod_timesql.py
@@ -17,7 +17,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 from httpx import ReadTimeout, RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack
 from wapitiCore.language.vulnerability import Messages, _

--- a/wapitiCore/attack/mod_wapp.py
+++ b/wapitiCore/attack/mod_wapp.py
@@ -19,7 +19,7 @@ import json
 import os
 
 from httpx import RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack
 from wapitiCore.wappalyzer.wappalyzer import Wappalyzer, ApplicationData, ApplicationDataException

--- a/wapitiCore/attack/mod_wp_enum.py
+++ b/wapitiCore/attack/mod_wp_enum.py
@@ -2,7 +2,7 @@ import re
 import json
 from os.path import join as path_join
 
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.net.web import Request
 from wapitiCore.attack.attack import Attack

--- a/wapitiCore/attack/mod_xss.py
+++ b/wapitiCore/attack/mod_xss.py
@@ -20,7 +20,7 @@ from os.path import join as path_join
 from configparser import ConfigParser
 
 from httpx import ReadTimeout, RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack, Mutator, PayloadType, random_string_with_flags
 from wapitiCore.language.vulnerability import Messages, _

--- a/wapitiCore/attack/mod_xxe.py
+++ b/wapitiCore/attack/mod_xxe.py
@@ -23,7 +23,7 @@ from configparser import ConfigParser
 from os.path import join as path_join
 
 from httpx import ReadTimeout, RequestError
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack.attack import Attack, FileMutator, Mutator, PayloadReader, Flags
 from wapitiCore.language.vulnerability import Messages, _

--- a/wapitiCore/main/log.py
+++ b/wapitiCore/main/log.py
@@ -1,0 +1,26 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# This file is part of the Wapiti project (https://wapiti.sourceforge.io)
+# Copyright (C) 2006-2021 Nicolas SURRIBAS
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+from loguru import logger as logging
+
+logging.remove()
+logging.level("RED", no=45, color="<red>")
+logging.level("ORANGE", no=35, color="<yellow>")
+logging.level("GREEN", no=22, color="<green>")
+logging.level("BLUE", no=21, color="<blue>")

--- a/wapitiCore/main/wapiti.py
+++ b/wapitiCore/main/wapiti.py
@@ -39,7 +39,6 @@ from typing import AsyncGenerator
 import browser_cookie3
 import httpx
 from httpx import RequestError
-from loguru import logger as logging
 
 from wapitiCore.language.language import _
 from wapitiCore.definitions import anomalies, additionals, vulnerabilities, flatten_references
@@ -49,14 +48,9 @@ from wapitiCore.net.web import Request
 from wapitiCore.report import get_report_generator_instance, GENERATORS
 from wapitiCore.net.sql_persister import SqlPersister
 from wapitiCore.moon import phase
+from wapitiCore.main.log import logging
 
 from wapitiCore.attack import attack
-
-logging.remove()
-logging.level("RED", no=45, color="<red>")
-logging.level("ORANGE", no=35, color="<yellow>")
-logging.level("GREEN", no=22, color="<green>")
-logging.level("BLUE", no=21, color="<blue>")
 
 BASE_DIR = None
 WAPITI_VERSION = "Wapiti 3.0.5"

--- a/wapitiCore/net/crawler.py
+++ b/wapitiCore/net/crawler.py
@@ -31,7 +31,7 @@ import functools
 from time import sleep
 from typing import Tuple, List
 import asyncio
-from loguru import logger as logging
+from wapitiCore.main.log import logging
 
 # Third-parties
 import httpx

